### PR TITLE
Upgrade to v7.1 database prep details

### PR DESCRIPTION
**Pull request by cwarnermm**
Documented v7.1 upgrade database preparation details

### CODE REVIEW

File: source/upgrade/upgrading-mattermost-server.rst
Comment (agnivade) [inline source/upgrade/upgrading-mattermost-server.rst:29]: Can you clarify from where you got the info on reduced downtime? The downtime will be the same however way the upgrade is done. This is just to facilitate the customer to easily schedule a maintenance operation without needing to upgrade Mattermost.

Comment (cwarnermm) [reply source/upgrade/upgrading-mattermost-server.rst:29]: @agnivade - See [this conversation](https://community.mattermost.com/private-core/pl/3yfqzushmff3b8ekx99w3h66xe) for details.

Comment (plant99) [reply source/upgrade/upgrading-mattermost-server.rst]: Hey Agniva, Carrie, not sure if I put the text correctly.

What I meant is, pre-running a migration only blocks Reactions table, and other functionalities will work just fine. The update query on Reactions takes 4/5th of the migration time.

If mm-server start is blocked by migrations, then no functionality is available for the full migration time.

Hope I clarified things.

Comment (agnivade) [reply source/upgrade/upgrading-mattermost-server.rst]: Thanks Shivashis. Carrie - let's change the sentence to "Customers wanting to run the SQL queries before-hand can run these queries".

The fact that the migrations will obtain a lock can be mentioned at the beginning.

> so users' reactions posted during this time won't be reflected in the database.

This is not entirely correct. The reactions will be posted, but it will be blocked until the migrations are complete.

I think we should decide on a fixed format for these migration release notes. We do the same thing for every release. It might be helpful to be consistent every time rather than come up with different text for each release.

Comment (cwarnermm) [reply source/upgrade/upgrading-mattermost-server.rst]: Thanks, @plant99 -- the interpretation error is all mine. I appreciate your help clarifying the message we want to communicate!

File: source/upgrade/prepare-to-upgrade-mattermost.rst
Comment (nab-77) [inline source/upgrade/prepare-to-upgrade-mattermost.rst:30]: What are the impacts of the lock and not posting the reactions when running the queries not in a maintenance window?

**Review by mkraft — APPROVED** [review]

File: source/upgrade/upgrading-mattermost-server.rst
Comment (agnivade) [reply source/upgrade/upgrading-mattermost-server.rst]: @cwarnermm - Feel free to let me know if you have any questions. Happy to help :)

File: source/upgrade/prepare-to-upgrade-mattermost.rst
Comment (cwarnermm) [reply source/upgrade/prepare-to-upgrade-mattermost.rst:30]: @agnivade - Can you help answer Neil's question?

Comment (agnivade) [reply source/upgrade/prepare-to-upgrade-mattermost.rst:30]: @nab-77 - The impact is mentioned - "so users' reactions posted during this time won't be reflected in the database." Let me know if you have any more questions.

This is just like any other schema migration change that we do every release. Some migrations will lock the tables, some won't. When a migration locks a table, nothing gets written to that table during that time.

cc @plant99

Comment (cwarnermm) [reply source/upgrade/prepare-to-upgrade-mattermost.rst:30]: @nab-77 - Based on Agniva's input, I've also updated the introductory sentence to remove the specifics about reduced downtime.

**Comment by cwarnermm** [comment]
Reviewers: This PR should now be ready for final review. Thank you!

**Review by agnivade — APPROVED** [review]

File: source/upgrade/prepare-to-upgrade-mattermost.rst
Comment (cwarnermm) [inline source/upgrade/prepare-to-upgrade-mattermost.rst:30]:
```suggestion
Customers wanting to run the SQL queries before-hand can run the following queries before upgrading. This is fully backwards-compatible and obtains a lock on ``Reactions`` table, so users' reactions posted during this time won't be reflected in the database until the migrations are complete.
```

File: source/upgrade/prepare-to-upgrade-mattermost.rst
Comment (justinegeffen) [inline source/upgrade/prepare-to-upgrade-mattermost.rst:30]:
```suggestion
Customers wanting to run the SQL queries beforehand can run the following queries before upgrading. This is fully backwards-compatible and obtains a lock on ``Reactions`` table, so users' reactions posted during this time won't be reflected in the database until the migrations are complete.
```

File: source/upgrade/prepare-to-upgrade-mattermost.rst
Comment (justinegeffen) [inline source/upgrade/prepare-to-upgrade-mattermost.rst:30]:
```suggestion
You can run SQL queries before the upgrade to obtain a lock on ``Reactions`` table, so that users' reactions posted during this time won't be reflected in the database until the migrations are complete. This is fully backwards-compatible.
```

File: source/upgrade/prepare-to-upgrade-mattermost.rst
Comment (justinegeffen) [inline source/upgrade/prepare-to-upgrade-mattermost.rst:53]:
```suggestion
Self-hosted Mattermost customers using MySQL databases may notice the migration to release v7.0 taking longer than usual when there are a large number of rows in the ``FileInfo`` table. See the `important upgrade notes <https://docs.mattermost.com/upgrade/important-upgrade-notes.html>`__ documentation for details.
```

File: source/upgrade/prepare-to-upgrade-mattermost.rst
Comment (justinegeffen) [inline source/upgrade/prepare-to-upgrade-mattermost.rst:55]:
```suggestion
```

File: source/upgrade/prepare-to-upgrade-mattermost.rst
Comment (justinegeffen) [inline source/upgrade/prepare-to-upgrade-mattermost.rst:177]:
```suggestion
In `high availability <https://docs.mattermost.com/scale/high-availability-cluster.html>`__ environments, you should expect to schedule downtime for the upgrade to v6.0. Based on your database size and setup, the migration to v6.0 can take a significant amount of time, and may even lock the tables for posts which will prevent your users from posting or receiving messages until the migration is complete.
```

File: source/upgrade/prepare-to-upgrade-mattermost.rst
Comment (justinegeffen) [inline source/upgrade/prepare-to-upgrade-mattermost.rst:179]:
```suggestion
Ensure you review the `high availability cluster upgrade guide <https://docs.mattermost.com/scale/high-availability-cluster.html#upgrade-guide>`__, as well as the :doc:`important-upgrade-notes` to make sure you're aware of any actions you need to take before or after upgrading from your particular version.
```

File: source/upgrade/prepare-to-upgrade-mattermost.rst
Comment (justinegeffen) [inline source/upgrade/prepare-to-upgrade-mattermost.rst:123]:
```suggestion
      See the `Mattermost v6.0 DB schema migrations analysis <https://gist.github.com/streamer45/59b3582118913d4fc5e8ff81ea78b055#online-migration-mysql>`__ documentation for a sample execution and additional caveats.
```

File: source/upgrade/prepare-to-upgrade-mattermost.rst
Comment (justinegeffen) [inline source/upgrade/prepare-to-upgrade-mattermost.rst:131]:
```suggestion
      For a complete breakdown of PostgreSQL queries, as well as their impact and duration, see the `Mattermost v6.0 DB schema migrations analysis <https://gist.github.com/streamer45/59b3582118913d4fc5e8ff81ea78b055#postgresql-1>`__.
```

File: source/upgrade/prepare-to-upgrade-mattermost.rst
Comment (justinegeffen) [inline source/upgrade/prepare-to-upgrade-mattermost.rst:136]:
```suggestion
Customers upgrading from a release older than Mattermost v5.35 should expect extended downtime when upgrading to v6.0 due to the introduction of backend database architecture introduced in v5.35. This upgrade path isn't recommended for large installations. We recommend upgrading to the latest `Extended Support Release (ESR) <https://docs.mattermost.com/upgrade/extended-support-release.html>`__ first before upgrading to Mattermost v6.0. See the `Mattermost changelog <https://docs.mattermost.com/install/self-managed-changelog.html>`__ documentation for additional details.
```

File: source/upgrade/prepare-to-upgrade-mattermost.rst
Comment (justinegeffen) [inline source/upgrade/prepare-to-upgrade-mattermost.rst:110]:
```suggestion
      For a complete breakdown of MySQL queries, as well as their impact and duration, see the `Mattermost v6.0 DB schema migrations analysis <https://gist.github.com/streamer45/59b3582118913d4fc5e8ff81ea78b055#mysql-1>`__ documentation.
```

File: source/upgrade/prepare-to-upgrade-mattermost.rst
Comment (justinegeffen) [inline source/upgrade/prepare-to-upgrade-mattermost.rst:96]:
```suggestion
A large migration from v5.39 to v6.0 of 72+ million posts will take approximately 3 hours and 40 minutes to complete for a MySQL database. See the `Migration results analysis <https://gist.github.com/streamer45/868c451164f6e8069d8b398685a31b6e>`__ documentation for test specifications, data sizes, and test results.
```

File: source/upgrade/prepare-to-upgrade-mattermost.rst
Comment (justinegeffen) [inline source/upgrade/prepare-to-upgrade-mattermost.rst:94]:
```suggestion
A migration to v6.0 of 10+ million posts will take approximately 1 hour and 22 minutes to complete for a MySQL database. See the `Mattermost v6.0 DB schema migrations analysis <https://gist.github.com/streamer45/59b3582118913d4fc5e8ff81ea78b055>`__ documentation for test specifications, data sizes, and test results.
```

File: source/upgrade/prepare-to-upgrade-mattermost.rst
Comment (mkraft) [reply source/upgrade/prepare-to-upgrade-mattermost.rst]: @nab-77 The data isn't lost because it's not accepted in the first place, it'll be rejected by the database which you'll see reflected in the API response.

**Pull request merged**
